### PR TITLE
Date Time support for cal package

### PR DESCRIPTION
**Pull request by samlown**
* Now with basic Date-Time support

### CODE REVIEW

File: cal/date_time.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+				Hour:   hour,
+				Minute: minute,
+				Second: second,
+			},
+		},
+	}
+}
+
+// Now produces a new date time instance for the current UTC time.
+func Now() DateTime {
```
[the remark this answers is not in the document]
Comment (cavalle) [reply cal/date_time.go:45]: I think both `cal.Timestamp.Now()` and `cal.DateTime.Now()` would make sense if that happens.

File: build/schemas/cal/date-time.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://gobl.org/draft-0/cal/date-time",
+  "$ref": "#/$defs/DateTime",
+  "$defs": {
+    "DateTime": {
+      "type": "string",
+      "format": "date-time",
+      "title": "Date Time",
+      "description": "Civil date time in simplified ISO format with no time zone\ninformation, for example: 2021-05-26T13:45:00\n"
```
Comment (cavalle) [inline build/schemas/cal/date-time.json:10]: Do we want those line breaks in that description?

**Review by cavalle — APPROVED** [review]
LGTM! (just left a couple of minor comments)

File: cal/date_time.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (samlown) [reply cal/date_time.go]: You can't do that in Go :-) But `cal.DateTimeNow()` is possible. I've renamed now to `ThisSecond()` which I find quite amusing and of course is aligned with Date's `Today()`.

[the remark this answers is not in the document]
Comment (cavalle) [reply cal/date_time.go]: oh, of course! I wasn't thinking clearly about this method. I ❤️ `cal.ThisSecond()`

File: build/schemas/cal/date-time.json (hunk 2 above)
Comment (samlown) [reply build/schemas/cal/date-time.json]: I fixed this too. Any empty lines at the end of the here.Doc are deleted.

Comment (cavalle) [reply build/schemas/cal/date-time.json]: I also meant the line breaks in the middle. Those line breaks are there originally to force text wrapping in a monospaced presentation (i.e. a code editor), but this is the JSON schema and that description could be used, for example, to automatically generate HTML documentation and we won't want a `<br>` or a `<p>` to be added at those places.

Comment (samlown) [reply build/schemas/cal/date-time.json]: There isn't a great solution for that... Markdown will of course ignore single `\n` and they are not meaningful inside HTML, so I think they're safe to ignore.

Comment (cavalle) [reply build/schemas/cal/date-time.json]: It doesn't feel 100% safe to assume those `\n` will always be ignored. Arguably, if those line breaks are expected to be ignored, then they shouldn't be there in the first place. We could write our own helper to remove new lines unless there are more than one consecutively.

Comment (samlown) [reply build/schemas/cal/date-time.json]: While not the case yet, imagine if we had code samples inside the `here.Doc()`. You for sure don't want to remove the `\n` and I'm not sure what rules could be used to do that.

Comment (cavalle) [reply build/schemas/cal/date-time.json]: ok then

**Pull request merged**